### DISCUSSION TITLE
Bundled Code Reference Object

### DISCUSSION THREAD

**Discussion by clovis517** [CMSgov/price-transparency-guide#644] [Ideas]
@shaselton-usds 
For some payers the primary contributor to large file sizes is replication of a Bundled Code Array consisting of all surgical and non-surgical procedure codes and descriptions to be paid under the surgical bundle, repeated across every surgical procedure  code, across hundreds of facilities with individually negotiated surgical rates.  The replication can be avoided by using a Bundled Code Reference Object in place of the Bundled Code Array, as follows:

In-Network Object : Add the following alternative for reporting Bundled Code Arrays
```
   Field  : bundled_codes_references
   Name : Bundled Codes References
   Type : Array
   Definition : An array of code_bundle_ids defined in the Bundled Code Reference object.
   Required : No
```

Bundled Code Reference Object : Identical to Bundle Code Object but with a primary key
```
   Fileld : code_bundle_id
   Name : Code Bundle ID Number
   Type : Number
   Definition : The unique, primary key for the associated Code Bundle
   Required : Yes
```

**Comment by shaselton-usds** [comment]
Hi @clovis517 

This is interesting - thanks for bringing it up. Do you have an example of a file that you can point to?

I wonder how much of this duplication can be cut down by breaking these bundles into smaller in-network files and then referencing them in a ToC, but hard to say without seeing examples.

**Reply by clovis517** [reply]
@shaselton-usds thank you for responding.  When I posted I didn't realize this idea was previously proposed by @jleberre10 last year here [discussions/368](https://github.com/CMSgov/price-transparency-guide/discussions/368).

Here is an example in a very small 207mb file containing 2,898 occurrences of a 5,948 byte bundled code array, so 17mb or 8% of the total file size.  [https://transparency-in-coverage.bluecrossma.com/2023-01-01_Blue-Cross-and-Blue-Shield-of-Massachusetts-Inc_Blue-Care-Elect-Fully-Insured_in-network-rates.json.gz](https://transparency-in-coverage.bluecrossma.com/2023-01-01_Blue-Cross-and-Blue-Shield-of-Massachusetts-Inc_Blue-Care-Elect-Fully-Insured_in-network-rates.json.gz)

I will post a better example when I find one in the wild.  I have examined a test file where the total repeated array made up 11.6gb : 8600 byte bundled array * 5400 Surgery HCPCS/CPT Codes * 250 facilities

**Reply by shaselton-usds** [reply]
Thanks, I'll take a peek and get back to you.

**Reply by BonHadVan** [reply]
@shaselton-usds ~ Hi Scott ~ Wondering if you can provide any guidance on this scenario.  For the Surgical Bundles would it be acceptable to use CSTM-ALL and CSTM-00 in the Bundled Code Object to represent all codes related to the Surgical Bundle are included?  Or is it necessary to specifically list all CPT, HCPCS and Revenue Codes that are reimbursed as part of the surgical bundle?
